### PR TITLE
add custom gas validations

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-1942
Fixes RNBW-1943

additionally 

Fixes RNBW-1934
Fixes RNBW-1991

## What changed (plus any additional context for devs)

Adding alerts on top of the warnings we already have, when user press "done" or the custom fee panel is closed

## Dev checklist for QA: what to test

- Added an IS_TESTING check in the fees panel, I was trying to add mocks to e2e but I was waiting too much time already
- If there's any warning on the fees panel and you try to press "Done" or swipe down to close that sheet an alert should appear telling you to EDIT max base fee or miner tip or proceed anyways, if you press the edit button you should be sent back to the screen and the corresponding input should be focused, max base fee or miner tip

## Final checklist
[x] Assigned individual reviewers?
[x] Added labels?

### CODE REVIEW

File: src/components/expanded-state/custom-gas/FeesPanel.js [hunk 1]
```diff
@@ -443,56 +478,193 @@ export default function FeesPanel({
...
       setMaxBaseFeeWarning(null);
     }
   }, [maxBaseFee, currentBaseFee]);
 
   useEffect(() => {
     // validate not zero
-    if (!maxPriorityFee || isZero(maxPriorityFee)) {
-      setMaxPriorityFeeError('1 Gwei to avoid failure');
+    if (!maxPriorityFee) {
+      setMaxPriorityFeeError('0 Gwei to avoid failure');
```
Comment (brunobar79) [inline src/components/expanded-state/custom-gas/FeesPanel.js:509]: shouldn't be 1 here?

**Review by brunobar79 — APPROVED** [review]
LGTM

File: src/components/expanded-state/custom-gas/FeesPanel.js (hunk 1 above)
Comment (estebanmino) [reply src/components/expanded-state/custom-gas/FeesPanel.js:509]: ah, you can actually pass 0 as the miner tip

**Comment by estebanmino** [comment]
RNBW-1934 pOw https://recordit.co/ISq08eNDzh (animation of the warning is fixed in https://github.com/rainbow-me/rainbow/pull/2622)

**Comment by estebanmino** [comment]
Fixes RNBW-1991 PoW https://recordit.co/U8JvgMWPb9 current base fee changed but the explainer sheet didn't

File: src/components/expanded-state/custom-gas/FeesPanel.js (hunk 1 above)
Comment (jinchung) [reply src/components/expanded-state/custom-gas/FeesPanel.js:507]: hm did christian ask for this change? seems strange to me we would recommend 0 gwei even if it is possible

File: src/components/expanded-state/custom-gas/FeesPanel.js [hunk 2]
```diff
@@ -120,6 +158,8 @@ export default function FeesPanel({
   const [maxBaseFeeWarning, setMaxBaseFeeWarning] = useState(null);
   const [maxBaseFeeError, setMaxBaseFeeError] = useState(null);
 
+  const [userProcededOnWarnings, setUserProcededOnWarnings] = useState(false);
```
Comment (jinchung) [inline src/components/expanded-state/custom-gas/FeesPanel.js:161]: typo: `userProceededOnWarnings`

**Comment by jinchung** [comment]
I would expect the "Proceed anyway" button to behave the same way as hitting "Done" originally (leaving the fees panel sheet and letting you continue) - right now it behaves the same as "edit fees" button

**Comment by jinchung** [comment]
Is the sheet meant to hop around when the warnings show up? Feels like a bit of a jittery experience to me

File: src/components/expanded-state/custom-gas/FeesPanel.js [hunk 3]
```diff
@@ -91,23 +94,58 @@ const GAS_FEE_INCREMENT = 1;
...
 
+const LOWER_THAN_SUGGESTED = 'Lower than suggested';
+const HIGHER_THAN_NECESSARY = 'Higher than necessary';
+
+const ALERT_MESSAGE_HIGHER_MINER_TIP_NEEDED =
+  'Setting a higher miner tip is recommended to avoid issues.';
+const ALERT_MESSAGE_HIGHER_MAX_BASE_FEE_NEEDED =
+  'Setting a higher max base fee is recommended to avoid issues.';
+const ALERT_MESSAGE_LOWER =
+  'Double check that you entered the correct amount—you’re likely paying more than you need to!';
```
Comment (jinchung) [inline src/components/expanded-state/custom-gas/FeesPanel.js:105]: are these long dashes part of a spec? look kinda meh imo ?

Comment (estebanmino) [reply src/components/expanded-state/custom-gas/FeesPanel.js:105]: there's no design, this is what's currently on prod but i changed the `gas price` word

**Comment by estebanmino** [comment]
PoW https://recordit.co/2r6rLtYvN9

**Comment by estebanmino** [comment]
@jinchung animations are much much better now https://recordit.co/bqVIjBC3Wg

- Input focus is maintained to the "last focused" input. So if you're modifying the miner tip and go to an explainer, when you close the explainer the miner tip is going to be automatically focused
- Same happens with the validation alert, if you close the panel and alert will appear saying the miner tip needs to be modified, if you go to edit the miner tip input is going to be automatically focused. On this part the keyboard was jumping around but i fixed it.

File: src/hooks/useFeesPanelInputRefs.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    return currentFocusedInputHandle.current;
+  }, []);
+
+  const {
+    handleFocus,
+    lastFocusedInputHandle,
+    setLastFocusedInputHandle,
+  } = useMagicAutofocus(maxBaseFieldRef, findNextInput, true);
```
Comment (brunobar79) [inline src/hooks/useFeesPanelInputRefs.ts:34]: 🪄

**Comment by skylarbarrera** [comment]
@estebanmino animations look crisp now nice work. 👌 

Non-blocking comments, will do code review rn:

I also don't like the 0 miner tip warning, I think the question is do we want to force them to use 1 as a min or just nix the warning all together. 

as for the copy def looks kinda off, im gonna send some screens to Jackson and get his input

File: src/components/expanded-state/custom-gas/FeesPanel.js [hunk 5]
```diff
@@ -524,6 +584,123 @@ export default function FeesPanel({
...
+    maxPriorityFeeError,
+    maxPriorityFeeWarning,
+    selectedOptionIsCustom,
+    setCanGoBack,
+    userProceededOnWarnings,
+  ]);
+
+  useEffect(() => {
+    return function validate() {
+      validateGasParams.current?.();
```
Comment (skylarbarrera) [inline src/components/expanded-state/custom-gas/FeesPanel.js:691]: why wrap this in a function vs calling it directly/how is this called, and where are we returning too 🤔

**Review by skylarbarrera — APPROVED** [review]
💯

File: src/components/expanded-state/custom-gas/FeesPanel.js (hunk 5 above)
Comment (estebanmino) [reply src/components/expanded-state/custom-gas/FeesPanel.js:691]: this is to be able to run the validations on `unmount`, that way only when the component is being unmounted `function validate()` is going to be triggered

Comment (skylarbarrera) [reply src/components/expanded-state/custom-gas/FeesPanel.js:691]: yup just did some reading on this, thanks for the explanation 👌

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**